### DISCUSSION TITLE
Структура разделов сайта FHIR RU

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#86] [General]
Коллеги, предлагаю такую структуру для оглавления сайта FHIR RU - дерево заголовков семантической вики.

<img width="715" alt="image" src="https://user-images.githubusercontent.com/62525329/154850454-33206587-286a-463f-a495-73572d603a19.png">

**Comment by Vrodvi** [comment]
Нужно прояснить область применения данной семантической вики:
- предполагается одна вики на все российские руководства по реализации
- или предполагаются отдельные вики на каждое руководство по реализации?

Если данная вики предназначена конкретно под "корневое" руководство по реализации, то:
- специфические сценарии использования наверное нужно убрать - они должны будут стать частью отдельных более специфических руководств по реализации (далее - РпР), например, лабораторное РпР, РпР для ФРМО и т.п.

Если вики предназначается для ведения всех руководств, то в структуре не хватает уровня для этих руководств.
Например, то что сейчас обозначено как "сценарии использования" может стать входными точками для руководств по реализации вместе с корневым руководством (то что обозначено как Ru Core), т.е. можно переделать структуру следующим образом:
- FHIR RU
  - FHIR - общие сведения по FHIR. Тут видимо будут обозначены общие принципы ведения руководств по реализации (правила формирование идентификаторов, принцип работы со справочниками, особенности использования типов данных и т.п.)?
  - Руководства по реализации
    - Ru Core
    - ФРМО
    - Клинический диагноз
    - Лабораторный результат
  - Сценарии использования - эту страницу тоже оставить, но сделать из неё что-то вроде указателя - обозначить какие сценарии использования возможны и какими руководствами по реализации их можно покрыть (т.е. отсюда будут идти ссылки на руководства)
  - Сообщество - без изменений


Внутри веток с конкретными руководствами по реализации желательно договориться о более-менее единой структуре организации данных, например, как она организована в спецификации и многих зарубежных руководствах:
- узел руководства по реализации ("Клинический диагноз", "Лабораторный результат" и другие)
  - Getting Started - вводная часть руководства
  - Documentation - основная документация руководства. Внутри - специфические подразделы под каждое руководство, например:
    - диагноз
    - нозологическая единица
    - ИЛИ
    - лабораторный материал
    - протокол лабораторного исследования
    - и т.п.
  - FHIR Artifacts
    - Profiles
    - Extensions
    - Terminologies
    - и т.д.

**Reply by eugenekogan-spb** [reply]
> предполагается одна вики на все российские руководства по реализации
> или предполагаются отдельные вики на каждое руководство по реализации?

на мой взгляд, сейчас лучше одна вики - и для удобства создания и поддержки, и для удобства чтения

> в структуре не хватает уровня для этих руководств

Вопрос философский, стоит ли объединять RuCore и прикладные руководства общим заголовком. В моем предложении RuCore вынесен на верхний уровень просто в силу своей значимости и (возможно) частоте обращения при использовании сайта. С другой стороны, RuCore тоже является руководством по реализации "любых процессов в здравоохранении РФ", поэтому можно внести его под этот заголовок.

> Сценарии использования - эту страницу тоже оставить, но сделать из неё что-то вроде указателя - обозначить какие сценарии использования возможны и какими руководствами по реализации их можно покрыть (т.е. отсюда будут идти ссылки на руководства)

Представляется избыточным. Если хочется назвать узел "Руководства по реализации" (что терминологически корректно), то давайте так и сделаем, и в корне этого узла напишем про то, как и для чего они создаются.

> Внутри веток с конкретными руководствами по реализации желательно договориться о более-менее единой структуре организации данных

Поддерживаю.

**Reply by niquola** [reply]
Я думаю единое пространство имеет смысл до тех пор пока не будет веских причин разбивать

**Comment by OlegPenzin** [comment]
коллеги, на мой взгляд, принципиально, чтобы публикуемые ссылки на артефакты FHIR (StructureDefinition, ValueSet, CodeSystem)  вида, к примеру, http://fhir.ru/fhir/ru/core/StructureDefinition/ru-core-patient были статичными и не зависели от положения описания в дереве семантической вики

**Reply by eugenekogan-spb** [reply]
@niquola Это как реализовать?

**Reply by niquola** [reply]
Я думаю наоборот - скорее всего решение  - для каждого профайла или valueset иметь одноименную страничку в zendoc. Позицию для ссылок мы можем закрепить соглашением.

**Reply by niquola** [reply]
Также можем сделать проверку что такая страница всегда существует

**Reply by Vrodvi** [reply]
Давайте рассмотрим на конкретном примере.

Предположим, что закрепится структура вида:

- FHIR RU
  - Руководства по реализации
    - Ru Core
      - FHIR Artifacts
        - Profiles
          - ru-core-patient 

Возможно ли, что в данном случае у профиля ru-core-patient:
- в zendoc'е будет путь http://fhir-ru.zendoc.me/Руководства по реализации/Ru Core/FHIR Artifacts/Profiles/ru-core-patient 
- но url будет http://fhir.ru/fhir/ru/core/StructureDefinition/ru-core-patient

Т.е. сможем ли мы закрепить за узлами zendoc'а "базовые" url, которые будут использоваться для формирования url профилей и других ресурсов?

- FHIR RU
  - Руководства по реализации
    - Ru Core
      - FHIR Artifacts
        - Profiles --> закрепить тут "базовый url" = "http://fhir.ru/fhir/ru/core/StructureDefinition/"
          - ru-core-patient --> следовательно тут будет url, построенный от базового, "http://fhir.ru/fhir/ru/core/StructureDefinition/ru-core-patient"

--- 

Или путь " http://fhir-ru.zendoc.me/" - это только временное решение и в дальнейшем сайт семантической вики "переедет" на fhir.ru и тогда можно будет использовать url пути, формируемые внутри вики:

- FHIR RU --> http://fhir.ru/fhir/ru
  - Руководства по реализации
    - Ru Core --> http://fhir.ru/fhir/ru/core/
      - FHIR Artifacts
        - Profiles --> "http://fhir.ru/fhir/ru/core/StructureDefinition/"
          - ru-core-patient --> "http://fhir.ru/fhir/ru/core/StructureDefinition/ru-core-patient"
?

**Reply by Vrodvi** [reply]
@niquola Николай, прокомментируйте, пожалуйста.

**Reply by niquola** [reply]
Да мы переедем на fhir.ru!

Я думаю что мы подгоним пути в вики и урлы профайлов:
Например страничка документации пациента в базовом FHIR - http://hl7.org/fhir/patient.html
А url в SD Пациента - https://hl7.org/fhir/StructureDefinition/Patient просто редиректит на http://hl7.org/fhir/patient.html
Я посмотрю насколько обязательно следовать паттерну StructureDefinition/?profile - если нет то можно просто
сделать их одинаковыми - те fhir.ru/core/Patient и будет url для профиля - либо сделаем такой же redirect

**Reply by niquola** [reply]
Я бы складывал IG на первом уровне навигации:

```
fhir-ru
    manifest
    working-groups/
    core (IG)/
        intro
        scenarios
        Patient
        Organization
        ...
        changelog
    labs (IG)/
    diagnoses (IG)/
    terms/
        
```

**Reply by niquola** [reply]
Для IG можем завести иконки - будет явно указывать что эта папочка iG -   📦 core  📦 labs

**Reply by Vrodvi** [reply]
@niquola 
Николай, 
в связи с тем что взгляды на структуру разделов вики уже, хоть и незначительно, но различаются, а в последующем также не исключена модификация\перенос разделов и\или добавление\удаление промежуточных уровней, 
то хотелось бы обсудить вопрос стабильности присвоенных артефактам URL адресов.

Хотелось бы получить подтверждение, что присвоенные объектам адреса не будут изменяться со временем при изменении структуры разделов вики.

Так, например, хотелось бы чтобы для корневого профиля пациента "ru-core-patient" был закреплён стандартный адрес "http://fhir.ru/fhir/ru/core/StructureDefinition/ru-core-patient", вне зависимости от того в каком разделе вики расположен этот профиль.

ПРИМЕР:
Сначала создаём структуру разделов с промежуточными раделами:
- FHIR RU
  - Руководства по реализации
    - Ru Core
      - FHIR Artifacts
        - Profiles
          - ru-core-patient --> "http://fhir.ru/fhir/ru/core/StructureDefinition/ru-core-patient" (фиксированный адрес)

, затем меняем структуру, убирая лишние промежуточные разделы (ожидается, что URL профиля при этом не изменится):
- FHIR RU
  - Ru Core
    - ru-core-patient --> "http://fhir.ru/fhir/ru/core/StructureDefinition/ru-core-patient" (фиксированный адрес)

**Comment by niquola** [comment]
Я предлагаю делать вложенность как можно меньше:

* Манифест
* 👁️ Обзор
* 📦 Ядро (IG)
* * Сценарии
* * ⚛️ Пациент (SD)
* * * пример-1
*  * :atom: Идентификаторы пациента (VS)
* * ⚛️ Организация
* * change log
* 📦 Лаборатория (IG)
* 📦 Диагнозы (IG)
* 📦 СЭМД (IG)
* ☝️  Термины
* 👥 Рабочие Группы
*  * Административные Ресурсы
* * * минутки 2022...
* 👤  Люди
* 🏛️ Организации

**Comment by eugenekogan-spb** [comment]
Коллеги, [на встрече 1 ноября](http://fhir.ru/meetings.2022-11-01) мы решили разделить руководство RuCore, включающее профили для различных разделов информации (административной, клинической, лабораторной, финансовой), а также иные артефакты, с одной стороны, и руководство по СЭМД ПЛИ (или для всех СЭМД), с другой стороны.
Предлагаю следующую структуру сайта

* Сообщество FHIR Ru – заголовочная страница сообщества
**   Встречи
**   Рабочие группы
**   Организации
**   Люди
* Руководство Ru Core – заголовочная страница руководства RuCore (черновик сейчас [опубликован на сайте](http://fhir.ru/core) )
**   Административная информация – руководство по отражению российской административной информации
**   Клиническая информация – руководство по особенностям клинической информации в целом, в тексте могут упоминаться профили, расширения, терминологии
**   Информация о лабораторных исследованиях – руководство по особенностям лабораторной информации в целом
**   Финансовая информация – общие особенности отражения финансовой информации в РФ
**   Профили ресурсов – узловая страница со списком
***      Observation – RuCore Observation (?)
***      Observation – Clin Observation
***      Observation – Lab Observation
***      Patient – RuCore Patient
      Patient – Fin Patient
      Practitioner – RuCore Practitioner
**   Профили типов данных – наверное, просто узловая страница со списком
***      RuAddress
**   Расширения – узловая страница со списком
**   Системы идентификации
**   Системы кодирования и наборы значений
* СЭМД – руководства по реализации СЭМД на FHIR
**   СЭМД протокол лабораторного исследования
***      Профили
***      Расширения
***      Системы идентификации
***      Системы кодирования и наборы значений

* Методические материалы
* История изменений

**Reply by eugenekogan-spb** [reply]
Страничка "Административная информация" могла бы содержать следующий текст

### Российские особенности представления административной информации при передаче медицинских данных

Ru Core учитывает следующие особенности ведения административной информации в РФ
**Идентификаторы пациентов**
Для идентификации пациента могут быть использованы паспорт, ИНН, СНИЛС, полис ОМС, номер государственного свидетельства о рождении, номера определенных медицинских карт. 
Номер пациента в медицинской информационной системе может быть использован при определенных требованиях к system, с тем чтобы пара system-value была уникальной.
Номер медицинского свидетельства о рождении в настоящее время не рекомендован в качестве идентификатора, так как отсутствует единая для РФ система выдачи номеров.
**Идентификаторы организаций**
Для идентификации организации могут быть использованы ИНН, ОГРН.
При обмене информацией в рамках ЕГИСЗ необходимо использовать идентификатор организации (подразделения) по подсистеме ФРМО ЕГИСЗ.
**Идентификаторы случаев оказания медицинской помощи**
Случаи оказания медицинской помощи могут быть идентифицированы с помощью номеров соответствующих медицинских карт.
**Адресная система**
Предусмотрено расширение типа данных Address для ведения кодов по ФИАС
**Справочники НСИ ЕГИСЗ**
RuCore предусматривает использование следующих справочников НСИ ЕГИСЗ:
- пол пациента, 
- документы, удостоверяющие личность
- номенклатура медицинских услуг
- виды медицинской помощи
- формы медицинской помощи
- единицы измерения
- должности медицинских и фармацевтических работников
- виды нозологических единиц диагноза
- степень обоснованности диагноза
- справочник лабораторных материалов и образцов

RuCore не устанавливает обязательность использования перечисленных здесь атрибутов, но определяет порядок их отражения в FHIR.

**Reply by Vrodvi** [reply]
Мне не нравится что "руководства по реализации" ру-кора так сильно выделяются и отличаются по структуре подразделов.
Я бы, наоборот, унифицировал все виды руководств, как я описывал в комментарии https://github.com/fhir-ru/core/discussions/86#discussioncomment-2220486

Если какие-то руководства нужно выделить как базовые руководства ру-кора, то это можно сделать, например, единым префиксом в названии раздела руководства (скажем, все производные от ру-кора руководства именовать "ru-core: название_руководства"), например

```
Руководства по реализации
- Ru Core
- Ru Core: Клинический диагноз
- Ru Core: Лабораторный результат
- Ru Core: и т.д.
- ФРМО
- СЭМД: протокол лабораторного исследования
- СЭМД: и т.д.
```

**Reply by eugenekogan-spb** [reply]
@Vrodvi  как раз в предложенном варианте каждое руководство будет представлять собой один узел сайта, внутри которого будет однотипная структура разделов.
Эти разделы включают:
1) текстовые главы руководства, при наличии. В случае RuCore есть главы с административной информацией, клинической и лабораторной.
2) профили ресурсов
3) расширения
4) терминологии

То же будет применено к руководству по СЭМД ПЛИ.
